### ISSUE TITLE
PhET-iO Build fails during or after JSDoc phase

### ISSUE THREAD

**Issue by samreid** [phetsims/chipper#749]
From slack:

Chris Malley [6:28 PM]
CT appears to be running again.  Now all phet-io sims are failing with (e.g.) `Failure to grunt snapshot-1553470470256-phet-io/beers-law-lab: ... Fatal error: null` (edited) 

Sam Reid [7:18 PM]
I’ll take a look
I’m able to reproduce the problem locally, investigating further

**Comment by samreid** [comment]
JSDoc step says `Error: file doesnt exist:  ../phet-io/js/phet-io-initialize-globals.js`.  I added an error step above to help catch that.  Not sure why it can't find that file.

**Comment by samreid** [comment]
A whitespace sneaked in front of the last two paths:

```
generating jsdoc for phet-io from:  [ '../query-string-machine/js/QueryStringMachine.js',
  '../phet-io-wrappers/common/js/assert.js',
  '../phet-io-wrappers/common/js/WrapperTypes.js',
  '../tandem/js/PhetioIDUtils.js',
  '../phet-io-wrappers/common/js/Client.js',
  '../phet-io-wrappers/common/js/readFile.js',
  '../phet-io-wrappers/common/js/loadTemplate.js',
  ' ../phet-io/js/phet-io-initialize-globals.js',
  ' ../chipper/js/initialize-globals.js' ]
```

Perhaps the latest version of jsdoc is no longer tolerant of that.  I'll remove the extra whitespace.

**Comment by samreid** [comment]
I outputted the JSDoc command args and ran it manually.  Saw many warnings but no errors:

```
~/github/faradays-law$ node ../chipper/node_modules/jsdoc/jsdoc.js ../query-string-machine/js/QueryStringMachine.js ../phet-io-wrappers/common/js/assert.js ../phet-io-wrappers/common/js/WrapperTypes.js ../tandem/js/PhetioIDUtils.js ../phet-io-wrappers/common/js/Client.js ../phet-io-wrappers/common/js/readFile.js ../phet-io-wrappers/common/js/loadTemplate.js ../phet-io/js/phet-io-initialize-globals.js ../chipper/js/initialize-globals.js -c ../phet-io/doc/wrapper/jsdoc-config.json -d ../faradays-law/build/phet-io/doc/ --readme ../phet-io/doc/wrapper/phet-io-documentation_README.md
WARNING: The @private tag does not permit a value; the value will be ignored. File: QueryStringMachine.js, line: 154
WARNING: The @private tag does not permit a value; the value will be ignored. File: QueryStringMachine.js, line: 161
WARNING: The @public tag does not permit a value; the value will be ignored. File: Client.js, line: 39
WARNING: The @public tag does not permit a value; the value will be ignored. File: Client.js, line: 45
WARNING: The @private tag does not permit a value; the value will be ignored. File: Client.js, line: 1081
WARNING: The @private tag does not permit a value; the value will be ignored. File: Client.js, line: 1088
WARNING: The @public tag does not permit a value; the value will be ignored. File: initialize-globals.js, line: 599
WARNING: The @public tag does not permit a value; the value will be ignored. File: initialize-globals.js, line: 604
WARNING: The @public tag does not permit a value; the value will be ignored. File: initialize-globals.js, line: 606
WARNING: The @public tag does not permit a value; the value will be ignored. File: initialize-globals.js, line: 611
```

**Comment by samreid** [comment]
Perhaps the error is happing in `new jimp()`?  Here's a patch that gets things building again, but without thumbnails:

```
Index: js/grunt/buildRunnable.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/grunt/buildRunnable.js	(revision 35a01e9a29c342db00f214bfb106dc7f79437b3c)
+++ js/grunt/buildRunnable.js	(date 1553484166000)
@@ -293,13 +293,13 @@
       { width: 128, height: 84 },
       { width: 600, height: 394 }
     ];
-    for ( const size of thumbnailSizes ) {
-      grunt.file.write( `${buildDir}/${repo}-${size.width}.png`, await generateThumbnails( repo, size.width, size.height ) );
-    }
-
-    if ( brand === 'phet' ) {
-      grunt.file.write( `${buildDir}/${repo}-ios.png`, await generateThumbnails( repo, 420, 276, 90, jimp.MIME_JPEG ) );
-      grunt.file.write( `${buildDir}/${repo}-twitter-card.png`, await generateTwitterCard( repo ) );
-    }
+    // for ( const size of thumbnailSizes ) {
+    //   grunt.file.write( `${buildDir}/${repo}-${size.width}.png`, await generateThumbnails( repo, size.width, size.height ) );
+    // }
+    //
+    // if ( brand === 'phet' ) {
+    //   grunt.file.write( `${buildDir}/${repo}-ios.png`, await generateThumbnails( repo, 420, 276, 90, jimp.MIME_JPEG ) );
+    //   grunt.file.write( `${buildDir}/${repo}-twitter-card.png`, await generateTwitterCard( repo ) );
+    // }
   }
 };
Index: js/grunt/phet-io/copySupplementalPhetioFiles.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/grunt/phet-io/copySupplementalPhetioFiles.js	(revision 35a01e9a29c342db00f214bfb106dc7f79437b3c)
+++ js/grunt/phet-io/copySupplementalPhetioFiles.js	(date 1553484452000)
@@ -12,8 +12,8 @@
 
 // modules
 const ChipperStringUtils = require( '../../common/ChipperStringUtils' );
+const child_process = require( 'child_process' );
 const copyDirectory = require( '../copyDirectory' );
-const execute = require( '../execute' );
 const fs = require( 'fs' );
 const grunt = require( 'grunt' );
 const minify = require( '../minify' );
@@ -63,8 +63,8 @@
 
 // list of files to run jsdoc generation with. Assume that anything in the public lib file needs documentation
 const JSDOC_FILES = LIB_FILES.concat( [
-  ' ../phet-io/js/phet-io-initialize-globals.js',
-  ' ../chipper/js/initialize-globals.js'
+  '../phet-io/js/phet-io-initialize-globals.js',
+  '../chipper/js/initialize-globals.js'
 ] );
 const JSDOC_README_FILE = '../phet-io/doc/wrapper/phet-io-documentation_README.md';
 
@@ -323,8 +323,10 @@
   // First we tried to run the jsdoc binary as the cmd, but that wasn't working, and was quite finicky. Then @samreid
   // found https://stackoverflow.com/questions/33664843/how-to-use-jsdoc-with-gulp which recommends the following method
   // (node executable with jsdoc js file)
-  await execute( 'node', [ '../chipper/node_modules/jsdoc/jsdoc.js' ].concat(
+  child_process.spawnSync( 'node', [ '../chipper/node_modules/jsdoc/jsdoc.js' ].concat(
     JSDOC_FILES.concat( [ '-c', '../phet-io/doc/wrapper/jsdoc-config.json',
       '-d', `${buildDir}doc/`, '--readme', JSDOC_README_FILE ] ) ),
     { cwd: process.cwd(), shell: true } );
+
+  console.log( 'thats done' );
 };
\ No newline at end of file

```

**Comment by samreid** [comment]
@jbphet volunteered to assist.

**Comment by samreid** [comment]
@jbphet and I isolated the problem to this newly added code:

```js
    _7z.cmd( [ 'a', '-txz', allHTMLFilename + '.xz', allHTMLFilename ], err => {
      grunt.log.error( err );
      throw new Error( err );
    } );
```

If we comment out those lines, then build succeeds.  This code was introduced in https://github.com/phetsims/chipper/issues/746.  We'll make a note in that issue.

**Comment by samreid** [comment]
UPDATE: We also determined this error occurs when running `grunt --allHTML --brands=phet`, so this is probably failing for the app builds. 

Therefore, we could probably revert this work on LZMA to get CT clear again, however, @jbphet and I are not familiar enough with whether this is already being used to create sim app builds.  So we will need to confer with @mattpen

**Comment by mattpen** [comment]
The error in #746 was fixed in 073f8fa0dd865b634e8b10013bdb86c18a7ed6eb.  This shouldn't be failing anymore.

**Comment by samreid** [comment]
Looks fixed, thanks!  Closing.
